moto/s3/models.py

```diff
@@ -1645,6 +1647,13 @@ def create_bucket(self, bucket_name: str, region_name: str) -> FakeBucket:
     def list_buckets(self) -> List[FakeBucket]:
         return list(self.buckets.values())
 
+    def get_bucket_global(self, bucket_name: str) -> FakeBucket:
```

Comment:
Does it make sense to just rewrite `get_bucket` to include this? I think that would make the cross-account logic available for all methods.

If that's not possible, or it introduces complexity somewhere else, I'd be happy to merge this as is - just trying to understand the pros/cons of this approach 🙂